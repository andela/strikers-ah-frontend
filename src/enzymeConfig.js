import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
