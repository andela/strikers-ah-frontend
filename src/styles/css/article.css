@import url("https://fonts.googleapis.com/css?family=Raleway&display=swap");
.spiner {
  position: fixed;
  z-index: 1031;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #43c0f1; }

.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%; }
  .comments-container .comment-count {
    font-weight: bold;
    margin-left: 4%; }

html {
  height: 100%;
  margin: 0; }

body {
  font-family: "Arial", "Helvetica", "sans-serif";
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%; }

.container {
  width: 80%;
  margin: auto;
  overflow: hidden; }
  @media (max-width: 768px) {
    .container {
      width: 100%; } }

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid; }

.brand {
  margin-top: 10px;
  float: left; }

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0; }

header .icons ul li i {
  color: grey; }

header .icons {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end; }
  @media (max-width: 768px) {
    header .icons {
      display: none; } }

.brand .brandName {
  text-decoration: none; }

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9; }

.content {
  background-color: #e5e5e5;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px; }
  @media (max-width: 768px) {
    .content {
      margin: auto;
      margin-right: 5%;
      margin-left: 5%; } }

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center; }

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */ }
  @media (max-width: 768px) {
    .content .editor-tools .container .bar-tool {
      margin: auto;
      padding: 5px; } }

.content .editor-tools .container .bar-tool {
  margin-bottom: 100px;
  margin-left: 10%;
  margin-right: 10%; }

input[type='color'] {
  border: none; }

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  transform: rotate(90deg); }

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no; }

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  transition: all 0.3s linear 0s;
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px; }

.content .text-area {
  width: 100%; }

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray; }

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%; }

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px; }

.content .text-area .form #publish {
  cursor: pointer;
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px; }

nav.icons div {
  margin-right: 15px; }

nav.icons i {
  margin-top: 5px; }

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px; }

nav.icons div.search #search-button {
  margin-right: 10px; }

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  box-shadow: #808e9b;
  margin-right: 10px; }

.card:hover {
  cursor: pointer;
  text-decoration: none; }

.card > img {
  width: 100%;
  height: 50%; }

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize; }

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px; }

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px; }

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px; }

.contents .contentBody #articleBody {
  clear: both;
  float: left;
  margin-top: 15px;
  padding-left: 10%;
  padding-right: 10%; }

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
  line-height: 1.5em; }

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px; }

.article-icons .icon {
  margin-right: 10px;
  display: inline-block; }

.contentDescription a {
  text-decoration: none; }

.contentDescription a:link {
  color: #2f3640; }

.contentDescription a:visited {
  color: #2f3640; }

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px; }

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .articleDescription {
  height: 65px;
  margin-top: 5px; }

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%; }

.socialMedia {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer; }
  .socialMedia .facebookButton {
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
    .socialMedia .facebookButton.facebook {
      color: #3b5998;
      border-color: #3b5998; }
      .socialMedia .facebookButton.facebook:hover {
        background-color: #3b5998;
        border-color: #3b5998;
        color: #fff; }
    .socialMedia .facebookButton.twitter {
      color: #1da1f2;
      border-color: #1da1f2; }
      .socialMedia .facebookButton.twitter:hover {
        background-color: #1da1f2;
        border-color: #1da1f2;
        color: #fff; }
    .socialMedia .facebookButton.googleplus {
      color: #dd4b39;
      border-color: #dd4b39; }
      .socialMedia .facebookButton.googleplus:hover {
        background-color: #dd4b39;
        border-color: #dd4b39;
        color: #fff; }
    .socialMedia .facebookButton.linkedin {
      color: #0077b5;
      border-color: #0077b5; }
      .socialMedia .facebookButton.linkedin:hover {
        background-color: #0077b5;
        border-color: #0077b5;
        color: #fff; }
    .socialMedia .facebookButton.whatsapp {
      color: #128c7e;
      border-color: #128c7e; }
      .socialMedia .facebookButton.whatsapp:hover {
        background-color: #128c7e;
        border-color: #128c7e;
        color: #fff; }
  .socialMedia .twitterButton {
    color: #1da1f2;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .twitterButton:hover {
    background-color: #1da1f2;
    color: #fff; }
  .socialMedia .linkedinButton {
    color: #0077b5;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .linkedinButton:hover {
    background-color: #0077b5;
    color: #fff; }
  .socialMedia .whatsappButton {
    color: #128c7e;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .whatsappButton:hover {
    color: #fff;
    background-color: #128c7e; }
  .socialMedia .emailButton {
    color: #dd4b39;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .emailButton:hover {
    color: #fff;
    background-color: #dd4b39; }

.edit-history-section {
  margin-top: 10px; }
  .edit-history-section .hideButton {
    background: #fff;
    padding: 3px;
    color: #000;
    border-radius: 3px;
    cursor: pointer;
    margin-bottom: 3px;
    font-size: 20px; }

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5; }
  .history-section time {
    color: #9c9c9c;
    font-size: 12px;
    font-weight: bold; }

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39; }

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px; }

.likeButton:hover {
  color: #43c0f1; }

.buttonSpan {
  margin-right: 5px; }

.topImage input[type='file'] {
  cursor: inherit;
  display: block;
  min-height: 100%;
  min-width: 100%;
  opacity: 0;
  position: absolute;
  right: 0;
  text-align: right;
  top: 0;
  float: right; }

.topImage .file-upload {
  overflow: hidden;
  position: absolute;
  width: 100px;
  height: 100px; }

.topImage .file-upload .imageIcone {
  min-height: 100%;
  min-width: 100%;
  margin-left: 20%;
  color: #3b5998; }

.tags {
  padding: 10px;
  margin-left: 20px;
  background-color: #3b5998;
  color: #fff;
  border-radius: 5px;
  display: inline-block;
  cursor: pointer; }

#tags > input {
  border-radius: 5px;
  border: none;
  width: 178px;
  height: 36px;
  margin-bottom: 5px;
  padding: 10px;
  font-size: 15px;
  margin-right: 5px; }

#tags {
  width: 75%;
  float: left; }

#addTag {
  width: 10%;
  float: left; }

.tagList {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
  background-color: #4b4b4b;
  color: #fff;
  display: inline-block;
  padding: 5px;
  cursor: pointer; }
