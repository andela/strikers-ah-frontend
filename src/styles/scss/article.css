@import url('https://fonts.googleapis.com/css?family=Raleway&display=swap');
.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%;
}

.comments-container .comment-count {
  font-weight: bold;
  margin-left: 4%;
}

html {
  height: 100%;
  margin: 0;
}

body {
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%;
}

.container {
  width: 80%;
  margin: auto;
  overflow: hidden;
}

@media (max-width: 768px) {
  .container {
    width: 100%;
  }
}

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid;
}

.brand {
  margin-top: 10px;
  float: left;
}

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

header .icons ul li i {
  color: grey;
}

header .icons {
  margin-top: 10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media (max-width: 768px) {
  header .icons {
    display: none;
  }
}

.brand .brandName {
  text-decoration: none;
}

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9;
}

.content {
  background-color: #e5e5e5;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px;
}

@media (max-width: 768px) {
  .content {
    margin: auto;
    margin-right: 5%;
    margin-left: 5%;
  }
}

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center;
}

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-left: 30%;
  margin-right: 30%;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  -webkit-box-shadow: 0px 4px 4px #c4c4c4;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */
}

@media (max-width: 768px) {
  .content .editor-tools .container .bar-tool {
    margin: auto;
    padding: 5px;
  }
}

input[type='color'] {
  border: none;
}

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  -webkit-transform: rotate(90deg);
  transform: rotate(90deg);
}

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no;
}

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  -webkit-transition: all 0.3s linear 0s;
  transition: all 0.3s linear 0s;
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px;
}

.content .text-area {
  width: 100%;
}

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray;
}

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
}

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px;
}

.content .text-area .form #publish {
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px;
}

nav.icons div {
  margin-right: 15px;
}

nav.icons i {
  margin-top: 5px;
}

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px;
}

nav.icons div.search #search-button {
  margin-right: 10px;
}

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  -webkit-box-shadow: #808e9b;
  box-shadow: #808e9b;
  margin-right: 10px;
}

.card:hover {
  cursor: pointer;
  text-decoration: none;
}

.card > img {
  width: 100%;
  height: 50%;
}

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word;
}

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word;
}

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize;
}

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px;
}

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px;
}

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px;
}

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
}

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px;
}

.article-icons .icon {
  margin-right: 10px;
  display: inline-block;
}

.contentDescription a {
  text-decoration: none;
}

.contentDescription a:link {
  color: #2f3640;
}

.contentDescription a:visited {
  color: #2f3640;
}

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px;
}

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px;
}

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px;
}

.card .articleDescription {
  height: 65px;
  margin-top: 5px;
}

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%;
}

.socialMedia {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.socialMedia .facebookButton {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .facebookButton.facebook {
  color: #3b5998;
  border-color: #3b5998;
}

.socialMedia .facebookButton.facebook:hover {
  background-color: #3b5998;
  border-color: #3b5998;
  color: #fff;
}

.socialMedia .facebookButton.twitter {
  color: #1da1f2;
  border-color: #1da1f2;
}

.socialMedia .facebookButton.twitter:hover {
  background-color: #1da1f2;
  border-color: #1da1f2;
  color: #fff;
}

.socialMedia .facebookButton.googleplus {
  color: #dd4b39;
  border-color: #dd4b39;
}

.socialMedia .facebookButton.googleplus:hover {
  background-color: #dd4b39;
  border-color: #dd4b39;
  color: #fff;
}

.socialMedia .facebookButton.linkedin {
  color: #0077b5;
  border-color: #0077b5;
}

.socialMedia .facebookButton.linkedin:hover {
  background-color: #0077b5;
  border-color: #0077b5;
  color: #fff;
}

.socialMedia .facebookButton.whatsapp {
  color: #128c7e;
  border-color: #128c7e;
}

.socialMedia .facebookButton.whatsapp:hover {
  background-color: #128c7e;
  border-color: #128c7e;
  color: #fff;
}

.socialMedia .twitterButton {
  color: #1da1f2;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .twitterButton:hover {
  background-color: #1da1f2;
  color: #fff;
}

.socialMedia .linkedinButton {
  color: #0077b5;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .linkedinButton:hover {
  background-color: #0077b5;
  color: #fff;
}

.socialMedia .whatsappButton {
  color: #128c7e;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .whatsappButton:hover {
  color: #fff;
  background-color: #128c7e;
}

.socialMedia .emailButton {
  color: #dd4b39;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .emailButton:hover {
  color: #fff;
  background-color: #dd4b39;
}

.edit-history-section {
  margin-top: 10px;
}

.edit-history-section .hideButton {
  background: #fff;
  padding: 3px;
  color: #000;
  border-radius: 3px;
  cursor: pointer;
  margin-bottom: 3px;
  font-size: 20px;
}

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5;
}

.history-section time {
  color: #9c9c9c;
  font-size: 12px;
  font-weight: bold;
}

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39;
}

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px;
}

.likeButton:hover {
  color: #43c0f1;
}

.buttonSpan {
  margin-right: 5px;
}
/*# sourceMappingURL=article.css.map */
