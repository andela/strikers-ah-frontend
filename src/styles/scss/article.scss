$fonts: 'Arial', 'Helvetica', 'sans-serif';

$colors: (
  headerColor: #ffffff,
  brandNameColor: #00aef9,
  contentBackGround: #e5e5e5,
  buttonColor: #808e9b,
);

@mixin textArea {
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: $fonts;
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
}

$mobileView: 768px;

@mixin mobile {
  @media (max-width: #{$mobileView}) {
    @content;
  }
}

html {
  height: 100%;
  margin: 0;
}

body {
  font-family: $fonts;
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%;
}

.container {
  width: 80%;
  margin: auto;
  overflow: hidden;

  @include mobile {
    width: 100%;
  }
}

header {
  background-color: map-get($map: $colors, $key: headerColor);
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid;
}

.brand {
  margin-top: 10px;
  float: left;
}

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

// header .icons ul li{
//     float: left;
//     display: inline;
//     padding: 0 20px 0 20px;
// }
header .icons ul li i {
  color: grey;
}

header .icons {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;

  @include mobile {
    display: none;
  }
}

.brand .brandName {
  text-decoration: none;
}

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9;
}

.content {
  background-color: map-get($map: $colors, $key: contentBackGround);
  align-content: center;
  margin-right: 100px;
  margin-left: 100px;

  @include mobile {
    margin: auto;
    margin-right: 5%;
    margin-left: 5%;
  }
}

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center;
}

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-left: 30%;
  margin-right: 30%;
  margin-bottom: 10px;
  background-color: map-get($map: $colors, $key: headerColor);
  border-radius: 8px;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;

  /* position: absolute; */
  @include mobile {
    margin: auto;
    padding: 5px;
  }
}

input[type='color'] {
  border: none;
}

// .content .editor-tools .container .bar-tool {
//     margin: auto;
//     padding: 0 15px 0 15px;

//     @include mobile {
//         padding: 0 5px 0 5px;
//     }
// }

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  transform: rotate(90deg);
}

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no;
}

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  transition: all 0.3s linear 0s;
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px;
}

.content .text-area {
  width: 100%;
}

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray;
}

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  @include textArea;
}

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  @include textArea;
  min-height: 800px;
}

.content .text-area .form #publish {
  border: none;
  background: map-get($map: $colors, $key: buttonColor);
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px;
}

nav.icons div {
  margin-right: 15px;
}

nav.icons i {
  margin-top: 5px;
}

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px;
}

nav.icons div.search #search-button {
  margin-right: 10px;
}
