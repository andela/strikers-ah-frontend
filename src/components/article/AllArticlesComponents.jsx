/* eslint-disable class-methods-use-this */
/* eslint-disable no-return-assign */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable no-unused-vars */
/* eslint-disable react/destructuring-assignment */
import React, { Component, Fragment } from 'react';
import dotenv from 'dotenv';
import { Redirect } from 'react-router-dom';
import sweeetAlert from 'sweetalert';
import decodeToken from 'jwt-decode';
import reactParser from 'html-react-parser';
import { connect } from 'react-redux';
import Header from './Header';
import '../../styles/css/article.css';
import samplePic from '../../styles/img/headBook.jpeg';
import {
  getAllArticles,
  getOneArticle,
  deleteArticle,
} from '../../redux/actions/articleAction';
import Spinner from './Spinner';
import MessageDisplay from './MessageDisplay';

dotenv.config();

const { REACT_APP_FRONTENT } = process.env;
/**
 *@author: Innocent Nkunzi
 * @returns {*} Articleform
 */
export class AllArticles extends Component {
  /**
   *
   * @param {*} props
   */
  state = {};

  /**
   *@author: Innocent Nkunzi
   * @returns {*} Articleform
   * @param {*} slug
   */
  componentDidMount() {
    this.props.getAllArticles();
  }

  /**
   *@author: Innocent Nkunzi
   * @returns {*} Articleform
   * @param {*} slug
   */
  handleDelete = slug => {
    if (slug) {
      sweeetAlert({
        title: 'Are you sure?',
        text: 'Are you sure you want to delete this article?',
        icon: 'warning',
        buttons: true,
        dangerMode: true,
      })
        .then(willDelete => {
          if (willDelete) {
            this.props.deleteArticle(slug);
            sweeetAlert('Article has been deleted!', {
              icon: 'success',
            });
          }
        })
        .catch(err => {
          sweeetAlert('Ooop something went wrong', err);
        });
    }
  };

  /**
   *
   * @param {*} item
   * @returns {*} image
   */
  handleImage = item => {
    const image1 = samplePic;
    const { image } = item;
    const image3 = image === 'null' ? image1 : image;
    return image3;
  };

  /**
   * @author Innocent Nkunzi
   * @returns {*} component
   */
  render() {
    const token = localStorage.getItem('token');
    let decodedToken;
    if (token) {
      decodedToken = decodeToken(token, { Header: true });
    }
    let allArticle;
    const {
      article: { allArticles, message },
    } = this.props;
    if (message !== undefined && message === 'Not article found for now') {
      return <Redirect to="/not-found" />;
    }
    if (allArticles !== undefined && allArticles.length !== 0) {
      const arr = allArticles;
      return (
        <div>
          <Header />
          <div className="container">
            <div className="allArticlecontents">
              {
                (allArticle = arr.map(item => (
                  <Fragment key={item.id}>
                    <div className="card">
                      <img
                        className="image"
                        alt="articleImage"
                        src={this.handleImage(item)}
                      />
                      <div className="contentDescription">
                        <a href={`/article/${item.slug}`}>
                          <div className="articleCardHeader">
                            <p>{reactParser(item.title)}</p>
                          </div>
                          <div className="articleDescription">
                            <p>{reactParser(item.description)}</p>
                          </div>
                        </a>
                        <div className="buttons">
                          {decodedToken !== undefined &&
                          decodedToken.id === item.authorid ? (
                            <div>
                              <button
                                type="submit"
                                className="redButton"
                                onClick={() => this.handleDelete(item.slug)}
                                id="submit-data"
                                data-test="submit-data"
                              >
                                Delete
                              </button>

                              <button type="submit" className="lightBlueButton">
                                <a href={`/articlesedit/${item.slug}/edit`}>
                                  Edit
                                </a>
                              </button>
                            </div>
                          ) : (
                            ''
                          )}
                        </div>
                      </div>
                    </div>
                  </Fragment>
                )))
              }
            </div>
          </div>
        </div>
      );
    }

    return (
      <div>
        <Spinner />
      </div>
    );
  }
}
const mapStateToprops = state => ({
  article: state.Article,
});

export default connect(
  mapStateToprops,
  { getAllArticles, getOneArticle, deleteArticle },
)(AllArticles);
