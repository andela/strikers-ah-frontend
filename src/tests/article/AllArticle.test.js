import React from 'react';
import { mount } from 'enzyme';
import '../../enzymeConfig';

import { AllArticles } from '../../components/article/AllArticlesComponents';

const token =
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6OCwiZmlyc3RuYW1lIjpudWxsLCJsYXN0bmFtZSI6bnVsbCwidXNlcm5hbWUiOiJmcmFuayIsImVtYWlsIjoiaGFyZnJhazNAZ21haWwuY29tIiwiaW1hZ2UiOm51bGwsImlhdCI6MTU1OTU2NjY4NCwiZXhwIjoxNTU5NjUzMDg0fQ.aW1TnCYURFsV38nRwrPMuOHJhr1yLtu1BdDhqfi1UkY';
localStorage.setItem('token', token);

const props = {
  slug: 'fjakslneralnlk',
  handleOpen: jest.fn(),
  getOneArticle: jest.fn(),
  handleDelete: jest.fn(),
  sweeetAlert: jest.fn(),
  decodeToken: jest.fn(),
  deleteArticle: jest.fn(),
  article: {
    state: {
      Article: [],
    },
    allArticles: [
      {
        id: 1,
        title: 'dkjkd',
        description: 'fghkffbdbjd',
        body: 'dhh jdjjbfd kdfbdcdsj sdbdnfd',
        authorid: 8,
        slug: 'hjdshjsfkjfj',
      },
    ],
  },

  getAllArticles: jest.fn(),
  jwtDecode: jest.fn(),
  decodedToken: {
    id: 8,
  },
  willDelete: jest.fn(),
};

const wrapper = mount(<AllArticles {...props} />);

describe('should test the  `AllArticles`', () => {
  test('should render the  `AllArticles` component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should test the  `handleDelete` component', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleDelete');
    wrapper.instance().forceUpdate();
    // const formInput = wrapper.find('#submit-data');
    const submitBtn = wrapper.find('[data-test="submit-data"]');
    submitBtn.simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
});
