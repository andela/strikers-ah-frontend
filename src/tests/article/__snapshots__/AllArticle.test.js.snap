// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the  \`AllArticles\` should render the  \`AllArticles\` component 1`] = `ShallowWrapper {}`;
